aeppl/mixture.py

```diff
@@ -300,6 +302,67 @@ def mixture_replace(fgraph, node):
     return [new_mixture_rv]
 
 
+@local_optimizer((Elemwise,))
+def switch_mixture_replace(fgraph, node):
+    r"""Identify mixture sub-graphs defined via a Switch Op and replace them with
+    a place-holder MixtureRV Op
+    """
+    try:
+        if not isinstance(node.op.scalar_op, Switch):
+            return None  # pragma: no cover
+    except AttributeError:
+        return None
+
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    old_mixture_rv = node.default_output()
+
+    if len(node.inputs) != 3:
+        return None  # graph structure should be [index, component1, component2]
+
+    mixing_indices = node.inputs[0]
+    mixture_res = node.inputs[1:]
```

Comment:
I believe that `switch` statements such as `at.switch(I_rv, X_rv, Y_rv)` **always** takes three arguments: the index, component 1 and component 2. Is this true? I currently define `mixture_res` as the `[X_rv, Y_rv]`.